### PR TITLE
Run sub logging demo

### PR DESCRIPTION
**Pull request by jedwards4b**
Convert buildlib.gptl to python and call it and the buildlib.pio as subroutines

Test suite: hand testing, scripts_regression_tests.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit 

Fixes:

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: src/build_scripts/buildlib.pio [hunk 1]
```diff
@@ -137,21 +131,25 @@ def _main_func(description):
...
 
 
-
 if (__name__ == "__main__"):
-    _main_func(__doc__)
+    if ("--test" in sys.argv):
+        test_results = doctest.testmod(verbose=True)
+        sys.exit(1 if test_results.failed > 0 else 0)
+
+    mbldroot, minstallpath, mcaseroot = parse_command_line(sys.argv, __doc__)
```
Comment (jgfouca) [inline src/build_scripts/buildlib.pio]: I think you still want to use a _main_func approach here. All these variables are being inserted into the global namespace.

File: src/build_scripts/buildlib.gptl [hunk 2]
```diff
@@ -1,23 +1,66 @@
...
+        cmd = "{} {}".format(gmake_cmd, gmake_opts)
+        output = run_cmd_no_fail(cmd)
+        logger.info(output)
+
+if (__name__ == "__main__"):
+    if ("--test" in sys.argv):
+        test_results = doctest.testmod(verbose=True)
+        sys.exit(1 if test_results.failed > 0 else 0)
+
+    mbldroot, minstallpath, mcaseroot = parse_command_line(sys.argv, __doc__)
```
Comment (jgfouca) [inline src/build_scripts/buildlib.gptl]: Still want a _main_func approach.

File: scripts/lib/CIME/utils.py [hunk 3]
```diff
@@ -33,11 +33,30 @@ def redirect_stderr(new_target):
...
     try:
         yield new_target
     finally:
         sys.stdout, sys.stderr = old_stdout, old_stderr
 
+@contextmanager
+def redirect_logger(new_target, logger_name):
+
+    ch = logging.StreamHandler(stream=new_target)
+    ch.setLevel(logging.DEBUG)
```
Comment (jgfouca) [inline scripts/lib/CIME/utils.py]: Can you rebase your branch so I don't see my changes?

**Comment by jedwards4b** [comment]
I knew something was wrong with main -thanks.

**Pull request merged**